Mindbox.Data.Linq/Mapping/AttributedMetaModel.cs

```diff
@@ -204,21 +158,19 @@ internal MetaTable GetTableNoLocks(Type rowType)
 			var attrs = GetTableAttributes(root, true);
 			if (!attrs.Any())
 			{
-				metaTables.Add(rowType, null);
+				metaTables.TryAdd(rowType, null);
 				return null;
 			}
 
-			if (!metaTables.TryGetValue(root, out table))
-			{
-				table = new AttributedMetaTable(this, attrs.First(), root);
-				foreach (var inheritanceType in table.RowType.InheritanceTypes)
-					metaTables.Add(inheritanceType.Type, table);
-			}
+			table = metaTables.GetOrAdd(root, type => new AttributedMetaTable(this, attrs.First(), root));
+
+			foreach (var inheritanceType in table.RowType.InheritanceTypes)
```

Comment:
@IharBury is it fine now?

Reply:
That may reduce performance as the derived meta types are constructed every time

Reply:
Do you want some syncronization when adding `AttributedMetaTable` for the same root type?
I don't think that this could be a problem because it can only hurt performance during initialization, and it doen't seem to be a huge performace hit.

Reply:
Hmm, these lines should not be invoked any more after the root is in the dictionary, you're right